src/main/java/frc/trigon/robot/motorsimulation/MotorSimulation.java

```diff
@@ -0,0 +1,83 @@
+package frc.trigon.robot.motorsimulation;
+
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.controls.VoltageOut;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import frc.trigon.robot.utilities.Conversions;
+
+public abstract class MotorSimulation {
+    private PIDController pidController;
+    private ProfiledPIDController profiledPIDController;
+    private double voltageCompensationSaturation;
+    private double
+            kS,
+            kG,
+            kV,
+            kA;
+    private double voltage = 0;
+
+    public void applyConfiguration(MotorSimulationConfiguration config) {
+        profiledPIDController = new ProfiledPIDController(config.kP, config.kI, config.kD, new TrapezoidProfile.Constraints(config.maxVelocity, config.maxAcceleration));
+        pidController = new PIDController(config.kP, config.kI, config.kD);
+        voltageCompensationSaturation = config.voltageCompensationSaturation;
+
+        this.kS = config.kS;
+        this.kG = config.kG;
+        this.kV = config.kV;
+        this.kA = config.kA;
+
+        pidController.enableContinuousInput(config.minimumContinuousOutput, config.maximumContinuousOutput);
```

Comment:
This should only be enabled if was configured in the configuration. If is enabled, the minimum will be the negative half of the conversions factor and the maximum will be the positive half.

Reply:
wdym negative half and positive half?

Reply:
Let's say if you're using revolutions, -0.5 and 0.5, or for degrees -180 and 180